File: backend-py/req/get_users.py
import requests
from dataclasses import dataclass
from typing import List
from config import STRAPI_URL

@dataclass
class User:
    id: int
    username: str
    email: str
    rolle: str

def get_users(jwt: str) -> List[User]:
    url = f"{STRAPI_URL}/users"
    headers = {
        "Authorization": f"Bearer {jwt}"
    }

    response = requests.get(url, headers=headers)
    response.raise_for_status()  # Raise an exception for HTTP errors

    users = response.json()
    user_list = [User(id=user['id'], username=user['username'], email=user['email'], rolle=user['rolle']) for user in users]

    return user_list

File: backend-rust/req/get_users.rs
use config::STRAPI_URL;
use crate::loginfn::User;

#[tokio::main]
pub async fn get_users(jwt: String) -> Result<Vec<User>, reqwest::Error> {
    let url = format!("{}{}", STRAPI_URL, "/api/users");

    let client = reqwest::Client::new();

    let res = client
        .get(&url)
        .header("Authorization", format!("Bearer {}", jwt))
        .send()
        .await?
        .json::<Vec<User>>()
        .await?;

    return Ok(res);
}